### PR TITLE
US2 Added endorsed attribute to the database

### LINKED ISSUE #36: US3 Add endorsed attribute to a topic/main post in the database
Effort: low
Acceptance Criteria: The database schema for a topic has an attribute to mark it as endorsed or not.

### PR DESCRIPTION
**Pull request by etong11**
Resolves #36 

**Schema**
Added an endorsed attribute (boolean) to a post in the redis database. This was done by adding to the database schema where post attributes were listed in:

- `public/openapi/components/schemas/PostObject.yaml`
- `public/openapi/read/topic/topic_id.yaml`
- `public/openapi/write/posts/pid.yaml`
- `public/openapi/write/posts/pid/replies.yaml`

I modeled whether the attribute was of type boolean or number (0 or 1) based off of other boolean attributes such as isMain and deleted.

**Creating a post**
Added endorsed as an attribute when creating a post in `src/posts/create.js`. Endorsed is false by default. The current code doesn't offer a way to mark this attribute as true in the front-end yet.

I also added endorsed in the field lists specified in `src/posts/summary.js` and `src/search.js` as the tests were failing at the api/search path, possibly because the endorsed attribute was not being taken from the database if it was not in the field list.

### CODE REVIEW

File: dump.rdb
Comment (ldlin1) [inline dump.rdb:1]: This file may not need to be committed, as it just keeps track of the current state of the database.

File: src/topics/posts.js
Comment (ldlin1) [inline src/topics/posts.js:1]: I think this file also doesn't need to be committed, as I believe it logs for testing purposes.

**Review by ldlin1 — CHANGES_REQUESTED** [review]
Removing the dump.rdb and src/topics/posts.js file might be best as they aren't directly related to the issue. Besides that, the code looks good! The changes will definitely help with endorsing/unendorsing posts

File: src/topics/posts.js
Comment (etong11) [reply src/topics/posts.js:1]: I think we should commit it as the change removes a console statement accidentally left from a previous commit.

File: dump.rdb
Comment (etong11) [reply dump.rdb:1]: Solved the issue by deleting dump.rdb in commit abcd3b98e70b9487f47f1e50991731b1aece965b

Comment (ldlin1) [reply dump.rdb:1]: Alright, looks good!

File: src/topics/posts.js
Comment (ldlin1) [reply src/topics/posts.js:1]: Alright, sounds reasonable.

**Review by ldlin1 — APPROVED** [review]
The changes from the previous commit look good, the code looks good to merge!

**Pull request merged**